### PR TITLE
Settings and helper functions implemented with some interfaces

### PR DESCRIPTION
**Pull request by KaiyuanLiu0**
I chose [storj-uplink ](https://pkg.go.dev/storj.io/uplink) as the lib. However, I found myself lacking background knowledge of go-storage. Therefore, this pr has some errors to be improved in the later work.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -20,17 +38,51 @@ func (s *Storage) list(ctx context.Context, path string, opt pairStorageList) (o
...
+	return meta
+}
+
+func (s *Storage) move(ctx context.Context, src string, dst string, opt pairStorageMove) (err error) {
 	panic("not implemented")
 }
 
 func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairStorageRead) (n int64, err error) {
-	panic("not implemented")
+	buckets := s.project.ListBuckets(ctx, nil)
```
Comment (Xuanwo) [inline storage.go:51]: Why do we need to `ListBuckets` before `DownloadObject`?

**Comment by Xuanwo** [comment]
Please fix the conflict.

**Comment by Xuanwo** [comment]
Hello, any updates here?

**Comment by KaiyuanLiu0** [comment]
I am busy these weekdays therefore no big progress. I will work on this
project full time during weekend to finish on time.

Xuanwo ***@***.***>于2021年8月27日 周五16:26写道：

File: service.toml [hunk 2]
```diff
@@ -1,11 +1,13 @@
 name = "storj"
 
 [namespace.storage]
+implement = ["copier", "mover", "direr"]
```
Comment (Xuanwo) [inline service.toml:4]: How remove all these `implement` so that we can focus on implementing `Storager`?

File: storage.go [hunk 3]
```diff
@@ -7,7 +7,25 @@ import (
 	. "github.com/beyondstorage/go-storage/v4/types"
 )
 
+func (s *Storage) copy(ctx context.Context, src string, dst string, opt pairStorageCopy) (err error) {
```
Comment (Xuanwo) [inline storage.go:10]: Let's implement them further.

**Comment by xxchan** [comment]
Hello, what's your next schedule?

**Pull request closed without merging**